espnet2/tasks/enh.py

```diff
@@ -342,56 +355,78 @@ def build_preprocess_fn(
     ) -> Optional[Callable[[str, Dict[str, np.array]], Dict[str, np.ndarray]]]:
         assert check_argument_types()
 
-        dynamic_mixing = getattr(args, "dynamic_mixing", False)
-        use_preprocessor = getattr(args, "use_preprocessor", False)
-
-        assert (
-            dynamic_mixing and use_preprocessor
-        ) is not True, (
-            "'dynamic_mixing' and 'use_preprocessor' should not both be 'True'"
-        )
-
-        if dynamic_mixing and train:
-            retval = DynamicMixingPreprocessor(
-                train=train,
-                source_scp=args.train_data_path_and_name_and_type[0][0],
-                num_spk=args.separator_conf["num_spk"],
-                dynamic_mixing_gain_db=getattr(args, "dynamic_mixing_gain_db", 0.0),
-                utt2spk=getattr(args, "utt2spk", None),
-            )
-        elif use_preprocessor:
-            retval = EnhPreprocessor(
-                train=train,
-                # NOTE(kamo): Check attribute existence for backward compatibility
-                rir_scp=args.rir_scp if hasattr(args, "rir_scp") else None,
-                rir_apply_prob=args.rir_apply_prob
-                if hasattr(args, "rir_apply_prob")
-                else 1.0,
-                noise_scp=args.noise_scp if hasattr(args, "noise_scp") else None,
-                noise_apply_prob=args.noise_apply_prob
-                if hasattr(args, "noise_apply_prob")
-                else 1.0,
-                noise_db_range=args.noise_db_range
-                if hasattr(args, "noise_db_range")
-                else "13_15",
-                short_noise_thres=args.short_noise_thres
-                if hasattr(args, "short_noise_thres")
-                else 0.5,
-                speech_volume_normalize=args.speech_volume_normalize
-                if hasattr(args, "speech_volume_normalize")
-                else None,
-                use_reverberant_ref=args.use_reverberant_ref
-                if hasattr(args, "use_reverberant_ref")
-                else None,
-                num_spk=args.num_spk if hasattr(args, "num_spk") else 1,
-                num_noise_type=args.num_noise_type
-                if hasattr(args, "num_noise_type")
-                else 1,
-                sample_rate=args.sample_rate if hasattr(args, "sample_rate") else 8000,
-                force_single_channel=args.force_single_channel
-                if hasattr(args, "force_single_channel")
-                else False,
-            )
+        use_preprocessor = getattr(args, "preprocessor", None) is not None
+
+        if train and use_preprocessor:
```

Comment:
Why is the preprocessor only used for training? The preprocessor may have their own rules for handling training and inference data.

Reply:
I looked at the usages of --use_preprocessor, it look like they were only used in training. So I changed in this logic. But we can keep it as it was.